app/models/accounting/transaction.rb

```diff
@@ -95,4 +112,27 @@ def lookup_currency_id
       project.currency_id
     end
   end
+
+  def calculate_loan_amounts
+    loan_id = self.project_id
+
+    if self.loan_transaction_type == 'disbursement'
+      self.update_column(:change_in_principal, amount)
+    end
+
+    if amount <= interest_balance
```

Comment:
I think you need to wrap this entire if/else block in a check to see whether it's a repayment. Disbursements still need to update the balances (way down the end) so you can't just return after you're finished checking disbursements, but as it stands you'd execute all of these lines for both and they're specific to repayments.